test/jdk/jdk/internal/util/random/RandomSupportTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * @test
+ * @bug 8326227
+ * @modules java.base/jdk.internal.util.random
+ * @summary Verify that RandomSupport methods behave as specified
+ * @run junit RandomSupportTest
+ */
+
+import org.junit.jupiter.api.*;
+import java.util.random.RandomGenerator;
+import jdk.internal.util.random.RandomSupport;
+import static org.junit.jupiter.api.Assertions.*;
+public class RandomSupportTest {
+    private static class WorstCaseRandomGenerator implements RandomGenerator {
+        boolean havePreviousOutput = false;
+        @Override
+        public long nextLong() {
+            if (havePreviousOutput) {
+                // gets shifted to 0x4000_0000_0000_0000, which puts us in the center of the sampling rectangle and
+                // ensures that we'll never choose a point under the curve
+                return Long.MIN_VALUE;
+            } else {
+                havePreviousOutput = true;
+                return Long.MIN_VALUE | 255; // need high value in last byte in order to skip the fast path
+            }
+        }
+    }
+    @Test
+    public void testNextExponentialSoftCapped() {
+        for (double max = 1.0; max < 10.0; max++) {
+            WorstCaseRandomGenerator rng = new WorstCaseRandomGenerator();
+            assertTrue(RandomSupport.computeNextExponentialSoftCapped(rng, max) >= max);
```

Comment:
Would you mind changing this to something like:
```java
double val = RandomSupport.computeNextExponentialSoftCapped(rng, max);
assertTrue(val >= max, val + " isn't >= " + max);

```
That way if this test fails for any reason, then we get the necessary details on what the computed value is.

Reply:
Done.